src/Tempest/View/src/Elements/ElementFactory.php

```diff
@@ -8,21 +8,39 @@
 use DOMElement;
 use DOMNode;
 use DOMText;
+use Tempest\Container\Container;
+use function Tempest\Support\str;
 use Tempest\View\Element;
-use Tempest\View\View;
+use Tempest\View\Renderers\TempestViewCompiler;
+use Tempest\View\ViewComponent;
+use Tempest\View\ViewConfig;
 
 final class ElementFactory
 {
-    public function make(View $view, DOMElement $node): ?Element
+    private TempestViewCompiler $compiler;
+
+    public function __construct(
+        private readonly ViewConfig $viewConfig,
+        private readonly Container $container,
+    ) {
+    }
+
+    public function setViewCompiler(TempestViewCompiler $compiler): self
```

Comment:
We need this setter to prevent a circular dependency between `ElementFactory` and `TempestViewCompiler`. Might be good if the container could handle this, but out of scope for this PR